### PR TITLE
[server] Usage-Based: Don't block slot operations on old Team Subscri…

### LINKED ISSUE #13105: [usage-based] Team Owners are no longer able to re-assign seats
Once a user is `usage-based`, they should no longer be able to signup for new subscriptions.
But they should still be able to modify their existing one's.

### PR DESCRIPTION
**Pull request by geropl**
…ptions

 1. Instead, we just block signin up for new Team Subscriptions.
 2. Fixes an issue with displaying Team Subscriptions to owners who have no seat with their own subscription

## Related Issue(s)

Fixes #13105 

## How to test
 - sign up for an old Chargebee Team Subscription
 - create a team with Gitpod in it's name, and upgrade to UBP
 - try to modify the old Chargebee-based Team Subscription (e.g., try to assign yourself a seat)
 - note how it works :heavy_check_mark: 

## Release Notes

```release-note
NONE
```

## Werft options:

- [x] /werft with-preview
- [x] /werft with-payment
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`

### CODE REVIEW

**Comment by AlexTugarev** [comment]
@geropl, I've no idea how to pass step 1 from the "how to test". 

> sign up for an old Chargebee Team Subscription

did you manage to create a team sub? tried several times, no errors logged in server, just nothing happens on the frontend.

**Comment by geropl** [comment]
@AlexTugarev sorry! Fixed the CB webhook setup, a re-send your last "subscription created" event. Should work now?

**Comment by AlexTugarev** [comment]
Screen Shot 2022-09-20 at 14 23 39

Once upgraded the GitpodX team to UBP, I've no longer access to the old-style team configuration anymore. BTW that also means, I've no way cancel the old team subs. 

> try to modify the old Chargebee-based Team Subscription (e.g., try to assign yourself a seat)

That's not possible. 🧱

**Comment by geropl** [comment]
Thx @AlexTugarev! Will fix this as well, and ping you once it actually works. Sorry for the double blocker :confused:

**Comment by geropl** [comment]
@AlexTugarev Fixed the display bug and tested, should work as expected now :relieved:

**Comment by AlexTugarev** [comment]
Screen Shot 2022-09-20 at 17 19 03

new issue. tried twice.

**Comment by AlexTugarev** [comment]
also, after creating Gitpod22 team, and trying to enable UBP for that team, with no luck as in prev comment, I end up with no access to the old-style team sub in my settings. 

Screen Shot 2022-09-20 at 17 20 43

**Comment by geropl** [comment]
We discovered that:
 - there was a temporary (?) issue during Ale's testes where a) customers got created but b) subscriptions not
 - there was a gap for user who already were `usage-based`, but did not upgrade to `paid`, yet. This got fixed with [5fbeed6](https://github.com/gitpod-io/gitpod/pull/13106/commits/5fbeed6d6698d606f86534a17516290f02dce0fc).

**Comment by AlexTugarev** [comment]
Seen `Assigned team subscription slot` in logs after successfully assigned myself on a seat of an old-style team sub.

File: components/server/ee/src/billing/billing-mode.ts [hunk 1]
```diff
@@ -140,26 +145,31 @@ export class BillingModesImpl implements BillingModes {
...
-            if (hasCbTeamSeat) {
+            if (hasCbTeamSeat || hasCbTeamSubscription) {
                 result.hasChargebeeTeamSubscription = true;
             }
             return result;
         }
+
+        if (hasUbbPaidTeam || hasUbbPersonal) {
+            // UBB is greedy: once a user has at least a paid team membership, they should benefit from it!
+            return usageBased();
```
Comment (AlexTugarev) [inline components/server/ee/src/billing/billing-mode.ts:163]: that all makes sense, and it works. it's just quite hard to read.

**Review by AlexTugarev — APPROVED** [review]
This works for me 🎉

File: components/server/ee/src/billing/billing-mode.ts (hunk 1 above)
Comment (geropl) [reply components/server/ee/src/billing/billing-mode.ts:163]: > it's just quite hard to read.

Yeah, that's the only downside I see here :confused: Still hoping all the complexity here is going to go away with Chargbee. :lotus_position:

**Pull request merged**